image/install-sriov-cni.sh

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+set -eu
+
+echo "Copying CNI plugins"
+cp -rv /opt/cni/bin/* /dest
```

Comment:
Given that this pod is presenting itself as persistent rather than the one-off that it is, whenever it gets restarted for some reason, it's going to re-copy these files.  Are they particularly large?  If so, we might consider caching a hash of the contents so that we only do so if they've changed.  Also, are there any concerns with overwriting them while in use?

Reply:
> Given that this pod is presenting itself as persistent rather than the one-off that it is, whenever it gets restarted for some reason, it's going to re-copy these files.

This container won't restart often - about the only thing I see leading to a container restart would be the host rebooting. We haven't seen container restarts on containerd upgrades or restarts. We won't see restarts on `kubectl drain` because it doesn't drain DaemonSet pods.

> Are they particularly large?

4.1M

> If so, we might consider caching a hash of the contents so that we only do so if they've changed.

Given how infrequent restarts are, I don't think the value gained is worth the added complexity.

> Also, are there any concerns with overwriting them while in use?

It might cause CNI to briefly fail when bringing a pod up, but Kubelet will retry shortly after, so I don't think it's much of a concern.